### PR TITLE
Support opening conversation activity by tapping notification

### PR DESCRIPTION
**Pull request by howawong**
- Add PushListenerService

Connects #15

### CODE REVIEW

File: app/src/main/java/io/skygear/chatdemo/PushListenerService.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        String conversationId = bundle.getString("conversation_id");
+
+        if (conversationId == null) {
+            Log.w(TAG, "conversationId is not provided.");
+            return;
+        }
+
+
+        mChatContainer.getConversation(conversationId, new GetCallback<Conversation>() {
```
Comment (carmenlau) [inline app/src/main/java/io/skygear/chatdemo/PushListenerService.java:86]: I think it is not a good option to fetch conversation when user received message, if another user keep sending messages to a room that have many people room. All the users got the message will fetch the conversation at the same time and cause server performance issue.

File: app/src/main/java/io/skygear/chatdemo/PushListenerService.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    public PushListenerService() {
+        this.random = new Random(new Date().getTime());
+        mSkygear = Container.defaultContainer(this);
+        mChatContainer = ChatContainer.getInstance(mSkygear);
+    }
+
+    private void createNotificationByConversation(String conversation, String title, String body) {
+        Intent mainActivityIntent = new Intent(this, ConversationsListActivity.class);
+        mainActivityIntent.addFlags(Intent.FLAG_ACTIVITY_BROUGHT_TO_FRONT);
```
Comment (carmenlau) [inline app/src/main/java/io/skygear/chatdemo/PushListenerService.java:42]: Would like to know what is `mainActivityIntent` for?

**Review by carmenlau — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/skygear/chatdemo/PushListenerService.java (hunk 2 above)
Comment (howawong) [reply app/src/main/java/io/skygear/chatdemo/PushListenerService.java:42]: Oh my bad, will delete that.

File: app/src/main/java/io/skygear/chatdemo/PushListenerService.java (hunk 1 above)
Comment (howawong) [reply app/src/main/java/io/skygear/chatdemo/PushListenerService.java:86]: You are right. However, shall I implement this in this demo project or chat SDK project?

**Comment by carmenlau** [comment]
I will close this PR, please open another after update. Thanks! :)

**Pull request closed without merging**